libut/Makefile

```diff
@@ -1,33 +1,41 @@
-OBJS=libut.a
-all: $(OBJS) 
+SHELL=/usr/sh
+VERSION=2.0.1
+MAJOR=2
+SOURCES=$(shell echo src/*.c)
+HEADERS=$(shell echo include/*.h)
+OBJS=$(SOURCES:.c=.o)
+all: libut.a libut.so.$(VERSION)
 INCDIR=./include
+LIBDIR=/usr/lib
 CFLAGS+=-I$(INCDIR)
-CFLAGS+=-Wall -Wextra
+CFLAGS+=-Wall -Wextra -Werror -fPIC
 CFLAGS+=-g
+CFLAGS+=$(OPTFLAGS)
 
-libut.a: libut.o utvector.o utmm.o ringbuf.o
+libut.a: $(OBJS)
 	ar r $@ $^
 
-libut.o: src/libut.c $(INCDIR)/libut.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utvector.o: src/utvector.c $(INCDIR)/utvector.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utmm.o: src/utmm.c $(INCDIR)/utmm.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-ringbuf.o: src/ringbuf.c $(INCDIR)/ringbuf.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
+libut.so.$(VERSION): $(OBJS)
+	$(CC) $(CFLAGS) -shared -Wl,-soname,libut.so.$(MAJOR) $^ -o $@
 
 .PHONY: clean tests install
 
 clean:
-	rm -f $(OBJS) *.o
+	rm -f *.o *.a *.so*
 	make -C tests clean
 
 tests: libut.a
 	make -C tests 
 
-install: libut.a
-	cp $< /usr/local/lib
+install: libut.a libut.so.$(VERSION)
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
+	mkdir -p $(DESTDIR)/usr/include/uthash
+	cp libut.a $(DESTDIR)$(LIBDIR)
+	cp libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so.$(MAJOR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so
+	cp ./include/*.h $(DESTDIR)/usr/include/uthash
```

Comment:
I'm treating this as a case of "extraordinary claims require extraordinary evidence": extraordinarily scary patches require extraordinarily convincing commit messages. The default is "let's leave /usr/include organized the way it is"; if you want to change that, I think you have to make some sort of argument in favor of the change and against the status quo. Otherwise, the status quo will (and should) win.

Also, if this is something that you think the OpenSUSE package maintainer would do anyway, why not let them do it, let them release the package and allow it to bake for a while, and then they can upstream the patch once it's proven to be the right thing to do? (I infer from your commit message that none of that has literally happened yet. My apologies if this *is* the upstreaming of an OpenSUSE patch. I found https://build.opensuse.org/package/show/devel:libraries:c_c++/uthash but those two patches are much simpler than this one and don't involve `make install`.)